lib/puppet/functions/conjur/client.rb

```diff
@@ -88,8 +88,8 @@ def variable_value account, id, token = nil
             ['variables', URI.encode_www_form_component(id), 'value']
           when 5
             raise ArgumentError, "account is required for v5" unless account
-            ['secrets', account, 'variable', id]
-          end.join('/')
+            ['secrets', account, 'variable', ERB::Util.url_encode(id)]
```

Comment:
how was this working for vars with slashes before? I have reduced confidence in the test cases below because they weren't erroring when this wasn't urlencoded before

Reply:
slashes aren't a problem as in both single and batch retrieval, we accept them as `/` and as `%2F`. So even when we don't encoded the id at all, a slash is fine.